### PR TITLE
[GUILD] Convert ahoy `ProgressTracker` to typescript

### PR DESCRIPTION
**Pull request by stefaandevylder**
### Description

Converts the `ProgressTracker` component to typescript.

### Manual check

Check storybook and compare it to the live one. It should work ok.

### CODE REVIEW

File: src/components/progressTracker/ProgressTracker.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  /** Whether or not all steps are completed */
+  done?: boolean;
+  /** The number of the step which is currently active */
+  currentStep: number;
+  /** The steps to display inside the progress tracker */
+  children?: ReactNode;
+  /** Color theme of the progress tracker. */
+  color: 'neutral' | 'mint' | 'aqua' | 'violet' | 'gold' | 'ruby' | 'teal';
+  /** Where to position the labels. Alternating allows for wider labels. */
+  labelPosition: 'top' | 'alternating' | 'bottom';
```
Comment (KristofColpaert) [inline src/components/progressTracker/ProgressTracker.tsx:19]: Q: are color, currentStep and labelPosition required? They have a default value, so they can be optional?

**Review by KristofColpaert — APPROVED** [review]

File: src/components/progressTracker/ProgressTracker.tsx (hunk 1 above)
Comment (stefaandevylder) [reply src/components/progressTracker/ProgressTracker.tsx:19]: True! Will edit ✅

**Review by KristofColpaert — DISMISSED** [review]

**Review by JorenSaeyTL — APPROVED** [review]

File: src/components/progressTracker/ProgressTracker.tsx [hunk 2]
```diff
@@ -10,13 +10,13 @@ export interface ProgressTrackerProps {
   /** Whether or not all steps are completed */
   done?: boolean;
   /** The number of the step which is currently active */
-  currentStep: number;
+  currentStep?: number;
   /** The steps to display inside the progress tracker */
   children?: ReactNode;
   /** Color theme of the progress tracker. */
-  color: 'neutral' | 'mint' | 'aqua' | 'violet' | 'gold' | 'ruby' | 'teal';
+  color?: 'neutral' | 'mint' | 'aqua' | 'violet' | 'gold' | 'ruby' | 'teal';
```
Comment (qubis741) [inline src/components/progressTracker/ProgressTracker.tsx:17]: Use  `COLORS` constant (2.a,ii)

**Review by qubis741 — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checking :hourglass:

**Review by driesd — DISMISSED** [review]

**Comment by qubis741** [comment]
Manual check failed :rotating_light: controls of story doesnt do anything

**Review by eniskraasniqi1 — APPROVED** [review]

**Review by qubis741 — DISMISSED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Review by JorenSaeyTL — DISMISSED** [review]

**Review by qubis741 — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Review by lorgan3 — APPROVED** [review]

**Pull request merged**